server/src/main/kotlin/com/xebia/functional/xef/server/http/routes/providers/RequestForwardingUtil.kt

```diff
@@ -0,0 +1,138 @@
+package com.xebia.functional.xef.server.http.routes.providers
+
+import com.xebia.functional.xef.server.models.Token
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import io.ktor.http.content.*
+import io.ktor.server.application.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.util.*
+import  io.ktor.http.URLBuilder
+import io.ktor.util.pipeline.*
+import io.ktor.utils.io.jvm.javaio.*
+
+private const val OAI_URL = "https://api.openai.com"
+
+/**
+ * Retrieves the path from the incoming request and
+ * combines it with the provider specific host url.
+ * As we are copying the API from the providers,
+ * everything including the path structure has follow the API contract.
+ */
+private fun buildProviderUrlFromRequest(request: ApplicationRequest) =
+    URLBuilder().takeFrom(OAI_URL).apply {
+        encodedPath = request.path()
+    }.build()
+
+@Deprecated("")
+internal suspend fun HttpClient.makeRequest(
```

Comment:
We are not using this method  and I don't understand why is deprecated without a description